content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md

```diff
@@ -5,6 +5,95 @@ weight=99
 
 <!--more-->
 
-This page is a stub! The CNB project is applying to [Google Season of Docs](https://developers.google.com/season-of-docs/docs/timeline) to receive support for improving our documentation. Please check back soon.
+The [buildpacks `exec.d` interface](https://github.com/buildpacks/spec/blob/main/buildpack.md#execd) allows buildpack authors to execute custom scripts when the application image is executed. This interface can be particularly useful for injecting dynamic behavior or environment variables into the runtime environment of an application.
 
-If you are familiar with this content and would like to make a contribution, please feel free to open a PR :)
+## Key Points:
+
+    1. Location and Naming: Scripts are placed in the `<layer>/exec.d/` directory within a lunch layer and must be executable. They can have any name.
+
+    2. Script Behavior:
+        * Environment Variables: Scripts can output environment variables in the form of key=value pairs. These variables are added to the application's runtime environment.
+        * File descriptor 3: The output should be directed to file descriptor 3, passed as the first argument to the executable, for environment variable settings. Any other output is ignored.
+        * Exit Code: The scripts should exit with a status code of `0` to indicate success. A non-zero exit code will indicate an error and prevent the application from launching.
+
+## Use Cases:
+* Dynamic Configuration: Inject configuration values that are determined at runtime.
+* Service Bindings: Configure environment variables based on bound services.
+
+## Implementation:
+* Write Scripts: Create executable scripts within the `<layer>/exec.d/` directory.
+* Set Permissions: Ensure scripts have the appropriate execute permissions (chmod +x).
+* Environment Variables: Use scripts to print `key="value"` pairs to the file descriptor passed as the first argument to the executable.
+
+### Example
+
+`exec.d` executables can be written in any language.  We provide examples in bash, Go and Python that inject the `EXAMPLE="test"` into the runtime environment.  It is important that environment variables are written to the open file descriptor passed as the first argument to the `exec.d` binary.
+
+A `bash` example looks as follows:
+```bash
+#!/bin/bash
+
+# Get the file descriptor from the first argument
+FD=$1
```

Comment:
```suggestion
FD=&3
```
I think this is how you would do this in bash? I would need to try it out to be sure

Reply:
The spec states that the FD is passed as the first argument https://github.com/buildpacks/spec/blob/main/buildpack.md#execd (or am I reading it wrong)?

Reply:
Hmm... I'm looking at

> 
Input | Description
-- | --
$0 | Absolute path of the executable
†FD 3 | A third open file descriptor

You can ignore the last two lines of the table (`<handle>` and the env var). They are for Windows only.